File: sdk/javascript/src/symbol/SharedKey.js
/* eslint-disable no-unused-vars */
import { KeyPair } from './KeyPair.js';
import { PublicKey, SharedKey256 } from '../CryptoTypes.js';
/* eslint-enable no-unused-vars */
import { deriveSharedKeyFactory } from '../SharedKey.js';
import { sha512 } from '@noble/hashes/sha512';

const deriveSharedKeyImpl = deriveSharedKeyFactory('catapult', sha512);

/**
 * Derives shared key from key pair and other party's public key.
 * @param {KeyPair} keyPair Key pair.
 * @param {PublicKey} otherPublicKey Other party's public key.
 * @returns {SharedKey256} Shared encryption key.
 */
const deriveSharedKey = (keyPair, otherPublicKey) => deriveSharedKeyImpl(keyPair.privateKey.bytes, otherPublicKey);

export { deriveSharedKey }; // eslint-disable-line import/prefer-default-export

File: sdk/python/symbolchain/symbol/SharedKey.py
from ..SharedKey import SharedKey as BasicSharedKey


class SharedKey(BasicSharedKey):
	@staticmethod
	def derive_shared_key(key_pair, other_public_key):
		"""Derives shared encryption key from private key and public key."""

		return BasicSharedKey._derive_shared_key(other_public_key.bytes, key_pair.private_key.bytes, b'catapult')
